tests/epics/test_pvi.py

```diff
@@ -144,3 +144,51 @@ async def test_device_create_children_from_annotations(
     assert device.device is block_2_device
     assert device.device.device is block_1_device
     assert device.signal_device is top_block_1_device
+
+
+@pytest.fixture
+def pvi_test_device_with_device_vectors_t():
+    """A fixture since pytest discourages init in test case classes"""
+
+    class TestDevice(Block2):
+        def __init__(self, prefix: str, name: str = ""):
+            self._prefix = prefix
+            create_children_from_annotations(self, device_vectors={"device_vector": 2})
+            super().__init__(name)
+
+        async def connect(
+            self, mock: bool = False, timeout: float = DEFAULT_TIMEOUT
+        ) -> None:
+            await fill_pvi_entries(
+                self, self._prefix + "PVI", timeout=timeout, mock=mock
+            )
+
+            await super().connect(mock=mock)
+
+    yield TestDevice
+
+
+async def test_device_create_children_from_annotations_with_device_vectors(
+    pvi_test_device_with_device_vectors_t,
+):
+    async with DeviceCollector(mock=True):
+        device = pvi_test_device_with_device_vectors_t("PREFIX:", name="test_device")
+
```

Comment:
```suggestion
    device = pvi_test_device_with_device_vectors_t("PREFIX:", name="test_device")
```

I believe from the rest of the code you probably don't want device to be connected here, only initialized?